### ISSUE TITLE
Errror while trying to draw visualization using open3d after running sapien render

### ISSUE THREAD

**Issue by j-thapa** [haosulab/SAPIEN#186]
after i initiate sapien ; in the same script if i try to draw visualization using open3D i get following error:

libGL error: MESA-LOADER: failed to open swrast: /usr/lib/dri/swrast_dri.so: cannot open shared object file: No such file or directory (search paths /usr/lib/x86_64-linux-gnu/dri:\$${ORIGIN}/dri:/usr/lib/dri, suffix _dri)
libGL error: failed to load driver: swrast
[Open3D WARNING] GLFW Error: GLX: Failed to create context: GLXBadFBConfig
[Open3D WARNING] Failed to create window
[Open3D WARNING] [DrawGeometries] Failed creating OpenGL window.
X Error of failed request:  BadWindow (invalid Window parameter)
  Major opcode of failed request:  15 (X_QueryTree)
  Resource id in failed request:  0x2c00026
  Serial number of failed request:  1518

**Comment by fbxiang** [comment]
It is very surprising that SAPIEN would have any effect on open3d since SAPIEN does not use OpenGL at all. And there is certainly no way to cause /usr/lib/dri/swrast_dri.so to go missing. Can you first verify this problem does not occur with open3d alone and is not caused some other libraries and provide a minimal example?
For example, does `import sapien` enough to cause the issue, or you need to create a scene or create a viewer?

**Comment by j-thapa** [comment]
Sorry for the late response; I wasnt working on this due to holidays and other stuff. 

I did some minor check and even import sapien causes this issue; if i import sapien or use code like from sapien.utils import Viewer I get the error. If i dont use this commands the open3d viewer run fine. Here is the demo code I used

'''
import sapien as sapien
from sapien.utils import Viewer
import numpy as np
import open3d as o3d


def main():
    # scene = sapien.Scene()  # Create an instance of simulation world (aka scene)
    # scene.set_timestep(1 / 100.0)  # Set the simulation frequency

    # # NOTE: How to build (rigid bodies) is elaborated in create_actors.py
    # scene.add_ground(altitude=0)  # Add a ground
    # actor_builder = scene.create_actor_builder()
    # actor_builder.add_box_collision(half_size=[0.5, 0.5, 0.5])
    # actor_builder.add_box_visual(half_size=[0.5, 0.5, 0.5], material=[1.0, 0.0, 0.0])
    # box = actor_builder.build(name="box")  # Add a box
    # box.set_pose(sapien.Pose(p=[0, 0, 0.5]))

    # # Add some lights so that you can observe the scene
    # scene.set_ambient_light([0.5, 0.5, 0.5])
    # scene.add_directional_light([0, 1, -1], [0.5, 0.5, 0.5])

    # viewer = scene.create_viewer()  # Create a viewer (window)

    # # The coordinate frame in Sapien is: x(forward), y(left), z(upward)
    # # The principle axis of the camera is the x-axis
    # viewer.set_camera_xyz(x=-4, y=0, z=2)
    # # The rotation of the free camera is represented as [roll(x), pitch(-y), yaw(-z)]
    # # The camera now looks at the origin
    # viewer.set_camera_rpy(r=0, p=-np.arctan2(2, 4), y=0)
    # viewer.window.set_camera_parameters(near=0.05, far=100, fovy=1)

    # while not viewer.closed:  # Press key q to quit
    #     scene.step()  # Simulate the world
    #     scene.update_render()  # Update the world to the renderer
    #     viewer.render()

          # Replace this with your actual 3D points
    points_world = np.random.rand(100, 3)  # Example: 100 random points in 3D space

    # Create an Open3D PointCloud object
    point_cloud = o3d.geometry.PointCloud()

    # Assign the points to the PointCloud object
    point_cloud.points = o3d.utility.Vector3dVector(points_world)

    # Optionally, add colors to the point cloud (e.g., random colors)
    colors = np.random.rand(points_world.shape[0], 3)
    point_cloud.colors = o3d.utility.Vector3dVector(colors)

    # Visualize the point cloud
    o3d.visualization.draw_geometries([point_cloud], window_name="3D Point Cloud")


if __name__ == "__main__":
    main()
'''

If I comment out ''' import sapien ''' and ''' from sapien.utils import Viewer ''' the visualization runs fine

**Comment by fbxiang** [comment]
Do you see any warnings from SAPIEN. For example, warnings about a broken driver installation where SAPIEN tries to workaround? One possibility is that your GPU graphics driver is not installed properly and SAPIEN tries to workaround this by providing missing EGL and Vulkan files when imported. And these files may interfere with open3d.

For example, one possibility is that previously open3d is using the CPU to render, but since SAPIEN provides these GPU files, open3d now thinks it is fine to use the GPU and uses a incorrectly configured GPU.

**Comment by fbxiang** [comment]
Also, your issue seems quite rare for a desktop environment. Are you using some special desktop setup such as x-forwarding, VNC, remote desktop, or docker?

**Comment by j-thapa** [comment]
yes, I am using forwarding to run ubuntu remotely on windows; maybe that's why am facing such issue.

I also find out that if I import open3d at first I don't get the error. So, its working fine for me now.

Thanks for the quick responses.
